test/cosign_test.py

```diff
@@ -3,48 +3,235 @@
+
+
+def test_image_signing(
+    payload_size: int,
+    payload_digest: str,
+    cfg_blob_size: int,
+    cfg_blob_digest: str,
+    public_key_1: str,
+    public_key_2: str,
+):
+    '''
+    Tests signing of an imaginary image multiple times in a row, starting without an existing
+    signature manifest and re-using the created manifest for subsequent signatures (append mode).
+
+    Scenarios:
+    1. Signing using `rsassa-pss` algorithm and `public_key_1`
+    2. Re-signing using same algorithm and key but different signature (as algorithm enforces)
+    3. Re-signing using `rsassa-pkcs1-v1-5` algorithm and `public_key_1`
+    4. Re-signing using `rsassa-pss` algorithm again but with `public_key_2` and changed signature
+    '''
+    manifest = oci.model.OciImageManifest(
+        config=oci.model.OciBlobRef(
+            digest=cfg_blob_digest,
+            mediaType='application/vnd.oci.image.config.v1+json',
+            size=cfg_blob_size,
+        ),
+        mediaType='application/vnd.oci.image.manifest.v1+json',
+        layers=[],
+        annotations={},
+    )
+
+    signature = 'rsassa-pss-signature-dummy'
+    signing_algorithm = model.signing_server.SigningAlgorithm.RSASSA_PSS
+
+    signed_manifest = cosign.sign_manifest(
+        manifest=manifest,
+        payload_size=payload_size,
+        payload_digest=payload_digest,
+        signature=signature,
+        signing_algorithm=signing_algorithm,
+        public_key=public_key_1,
+        on_exist=cosign.OnExist.APPEND,
+    )
+
+    assert len(signed_manifest.layers) == 1
+    signature_layer = signed_manifest.layers[0]
+    assert signature_layer.annotations.get(cosign._cosign_signature_annotation_name) == signature
+    assert signature_layer.size == payload_size
+    assert signature_layer.digest == payload_digest
+
+    signature = 'rsassa-pss-signature-dummy-changed'
+
+    manifest_resigned = cosign.sign_manifest(
+        manifest=dataclasses.replace(signed_manifest), # copy of previous manifest
+        payload_size=payload_size,
+        payload_digest=payload_digest,
+        signature=signature,
+        signing_algorithm=signing_algorithm,
+        public_key=public_key_1,
+        on_exist=cosign.OnExist.APPEND,
+    )
+
+    assert signed_manifest == manifest_resigned # nothing should have changed during re-signing
+
+    signature = 'rsassa-pkcs1-v1-5-signature-dummy'
+    signing_algorithm = model.signing_server.SigningAlgorithm.RSASSA_PKCS1_V1_5
+
+    manifest_different_algorithm = cosign.sign_manifest(
+        manifest=dataclasses.replace(manifest_resigned), # copy of previous manifest
+        payload_size=payload_size,
+        payload_digest=payload_digest,
+        signature=signature,
+        signing_algorithm=signing_algorithm,
+        public_key=public_key_1,
+        on_exist=cosign.OnExist.APPEND,
+    )
+
+    assert len(manifest_different_algorithm.layers) == 2
+    signature_layer = manifest_different_algorithm.layers[1]
```

Comment:
we might also re-check existing signatures are not change. but I think this should be good-enough like this

Reply:
As part of the integration test, we also check if the public key of the first signature is still able to verify the signature after appending a new one.